src/Server/OutputParser.cs

```diff
@@ -0,0 +1,150 @@
+//-----------------------------------------------------------------------------
+// <copyright file="OutputParser.cs" company="WheelMUD Development Team">
+//   Copyright (c) WheelMUD Development Team.  See LICENSE.txt.  This file is 
+//   subject to the Microsoft Public License.  All other rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------------
+
+using WheelMUD.Utilities;
+
+namespace WheelMUD.Server
+{
+    public static class OutputParser
+    {
+        //TODO: Optimize more
+        public static string NonAnsi(char[] buffer, int bufferPos, int wordWrapLength)
```

Comment:
It seems a maintenance risk to have to copies of essentially the same algorithm here.  For example, despite the code being fresh on mind, already the first fix applied (the index safety checks) were only applied to _one_ of the two.  I feel we should collapse the bulk of these together and modify the `ansiResult += AnsiHandler.ConvertCode(token);` part to work conditionally between the two modes (omitting them if useAnsi is false).

The NonAnsi version is also failing to put in the NewLines; that's one bit both should do consistently but isn't.  (UseANSI is always true right now, but I think it was meant to be determined at negotiations time, probably based on whether "ansi" was in the list of supported client modes or whatnot.  We don't successfully keep iterating through the client mode list right now.  Actually this aspect of the Telnet protocol is still a bit confusing to me.)  Anyway, our `UseANSI` bool refers to supporting the ANSI escaped sequences.  So regardless of UseANSI, I believe we should be sending `\r\n` for any NewLines.

Reply:
Moved into a single parse, add ansi bool to parse params.